src/passes/OptimizeInstructions.cpp

```diff
@@ -205,6 +215,182 @@ struct OptimizeInstructions
       if (isSymmetric(binary)) {
         canonicalize(binary);
       }
+    }
+
+    {
+      // TODO: It is an ongoing project to port more transformations to the
+      // match API. Once most of the transformations have been ported, the
+      // `using namespace Match` can be hoisted to function scope and this extra
+      // block scope can be removed.
+      using namespace Match;
+      Builder builder(*getModule());
+      {
+        // X == 0 => eqz X
+        Expression* x;
```

Comment:
may as well hoist and unify these `x, y` etc. declarations?

Reply:
I think it's actually better for readability to declare any necessary binder variables at the location where the pattern is used. That way, the reader only needs to read a small, contiguous block of code to fully understand what transformation is being applied. It also allows anyone adding a new pattern to use `x` with whatever type makes sense for that pattern.